lib/puppet/parser/compiler.rb

```diff
@@ -117,7 +117,7 @@ def assert_app_in_site(scope, resource)
           return if upstream.keys.map(&:type).include?('Site')
         end
       end
-      raise ArgumentError, "Application instances like '#{resource}' can only be contained within a Site"
+      raise ArgumentError, _("Application instances like '%{resource}' can only be contained within a Site") % { resource: resource }
```

Comment:
Can you add a note that `Site` is a key word in puppet and shouldn't be translated?